src/coreclr/jit/morph.cpp

```diff
@@ -11087,6 +11092,69 @@ GenTree* Compiler::fgOptimizeCast(GenTreeCast* cast)
     return cast;
 }
 
+//------------------------------------------------------------------------
+// fgOptimizeCastOfSmpOp: Optimizes the supplied GT_CAST of GTK_SMPOP tree.
+//
+// Arguments:
+//    cast - the cast tree to optimize
+//
+void Compiler::fgOptimizeCastOfSmpOp(GenTreeCast* cast)
```

Comment:
There are a few similar-looking transformations that exist. In particular: `fgMorphCastedBitwiseOp`, `optNarrowTree`. It would be good to see whether something of this could either be folded into something else or deleted.

In particular, in the past I've integrated this transformation into the latter (with good results), and here I wonder whether we still need the former.

Reply:
Definitely feels similar - would need to spend time on figuring out what things can be commonly shared between them.